ecommerce/serializers.py

```diff
@@ -354,3 +355,188 @@ class Meta:
             "user",
         ]
         depth = 1
+
+
+class OrderReceiptSerializer(serializers.ModelSerializer):
+    """
+    Serializer for extracting receipt info from an Order object
+    This hews pretty closely to the data format in xPro but modified a bit
+    for MITxOnline's data model.
+    """
+
+    lines = serializers.SerializerMethodField()
+    purchaser = serializers.SerializerMethodField()
+    coupon = serializers.SerializerMethodField()
+    order = serializers.SerializerMethodField()
+    receipt = serializers.SerializerMethodField()
```

Comment:
I think these are all (except `order`, but let's do it anyway for consistency) complex enough to warrant refactoring into their own serializers, which would also allow you to write smaller tests for each of those and then in the test for `OrderReceiptSerializer` just use something like `CouponSerializer(order.coupon).data` in your assertions since you'd have already tested that serializer independently.

An example of this is here in xPro: https://github.com/mitodl/mitxpro/blob/master/courses/serializers_test.py#L89-L94

Reply:
https://github.com/mitodl/mitxonline/pull/558/commits/fc631001025f20db1e8e711ddfa13b7a810e8c6a refactors these, except for the coupon one where I just re-used the existing DiscountRedemptionSerializer and changed the template to work with it instead.